### ISSUE TITLE
Errors when building with gcc-11

### ISSUE THREAD

**Issue by samuel40791765** [aws/aws-encryption-sdk-c#749]
### Problem:
There were some build errors when building against the default gcc-11 on Ubuntu20.04. Docker image used was the one owned by ESDK at `codebuild/ubuntu-latest-x64.Dockerfile`. Only change I made was to update the `AWS_CRT_CPP_VERSION` to `v0.17.28` and `AWS_SDK_CPP_VERSION` to `1.9.263` under `codebuild/bin/install-aws-deps.sh` to get updated dependencies. Without changing the git tag, the dependencies would fail to build as well.
gcc-10 works as expected.

```                                 
[8/131] /usr/bin/gcc -DAWS_COMMON_USE_IMPORT_EXPORT -DAWS_ENCRYPTION_SDK_EXPORTS -DAWS_ENCRYPTION_SDK_SHARED 
-DAWS_ENCRYPTION_SDK_TEST_SHARED -Daws_encryption_sdk_EXPORTS -I/home/ubuntu/workplace/aws-encryption-sdk-c/i
nclude -I/home/ubuntu/workplace/aws-encryption-sdk-c/noninst_include -I/home/ubuntu/workplace/aws-encryption-
sdk-c/build/generated/include -I/deps/install/include -isystem /deps/shared/install/include -O2 -g -DNDEBUG -
fPIC -Wall -Werror -Wreturn-type -Wextra -pedantic -Wno-long-long -Wno-missing-field-initializers -fvisibilit
y=hidden -std=gnu99 -MD -MT CMakeFiles/aws-encryption-sdk.dir/source/cipher_openssl.c.o -MF CMakeFiles/aws-en
cryption-sdk.dir/source/cipher_openssl.c.o.d -o CMakeFiles/aws-encryption-sdk.dir/source/cipher_openssl.c.o -
c /home/ubuntu/workplace/aws-encryption-sdk-c/source/cipher_openssl.c                                        
FAILED: CMakeFiles/aws-encryption-sdk.dir/source/cipher_openssl.c.o                                          
/usr/bin/gcc -DAWS_COMMON_USE_IMPORT_EXPORT -DAWS_ENCRYPTION_SDK_EXPORTS -DAWS_ENCRYPTION_SDK_SHARED -DAWS_EN
CRYPTION_SDK_TEST_SHARED -Daws_encryption_sdk_EXPORTS -I/home/ubuntu/workplace/aws-encryption-sdk-c/include -
I/home/ubuntu/workplace/aws-encryption-sdk-c/noninst_include -I/home/ubuntu/workplace/aws-encryption-sdk-c/bu
ild/generated/include -I/deps/install/include -isystem /deps/shared/install/include -O2 -g -DNDEBUG -fPIC -Wa
ll -Werror -Wreturn-type -Wextra -pedantic -Wno-long-long -Wno-missing-field-initializers -fvisibility=hidden
 -std=gnu99 -MD -MT CMakeFiles/aws-encryption-sdk.dir/source/cipher_openssl.c.o -MF CMakeFiles/aws-encryption
-sdk.dir/source/cipher_openssl.c.o.d -o CMakeFiles/aws-encryption-sdk.dir/source/cipher_openssl.c.o -c /home/
ubuntu/workplace/aws-encryption-sdk-c/source/cipher_openssl.c                                                
/home/ubuntu/workplace/aws-encryption-sdk-c/source/cipher_openssl.c: In function 'aws_cryptosdk_sig_get_privk
ey':                                                                                                         
/home/ubuntu/workplace/aws-encryption-sdk-c/source/cipher_openssl.c:358:36: error: 'tmparr' may be used unini
tialized [-Werror=maybe-uninitialized]                                                                       
  358 |     struct aws_byte_buf tmpbuf   = aws_byte_buf_from_array(tmparr, sizeof(tmparr));                  
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                   
In file included from /deps/shared/install/include/aws/common/encoding.h:9,                                  
                 from /home/ubuntu/workplace/aws-encryption-sdk-c/source/cipher_openssl.c:32:                
/deps/shared/install/include/aws/common/byte_buf.h:563:36: note: by argument 1 of type 'const void *' to 'aws
_byte_buf_from_array' declared here                                                                          
  563 | AWS_COMMON_API struct aws_byte_buf aws_byte_buf_from_array(const void *bytes, size_t len);           
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~                                           
/home/ubuntu/workplace/aws-encryption-sdk-c/source/cipher_openssl.c:354:19: note: 'tmparr' declared here     
  354 |     unsigned char tmparr[MAX_PUBKEY_SIZE * 2 + 2];                                                   
      |                   ^~~~~~                                                                             
cc1: all warnings being treated as errors
```

### Solution:

Could be a false positive. If so, suppressing the warning should fix it.

### Out of scope:

Is there anything the solution will intentionally NOT address?

[//]: #  (NOTE: If you believe this might be a security issue, please email aws-security@amazon.com instead of creating a GitHub issue. For more details, see the AWS Vulnerability Reporting Guide: https://aws.amazon.com/security/vulnerability-reporting/ )

**Comment by bitpup** [comment]
also for alpinelinux 3.16.0 / gcc 11

**Comment by alex-chew** [comment]
Resolved by #751.
